core/block/editor/state/position.go

```diff
@@ -180,23 +172,69 @@ func (s *State) moveFromSide(target, parent simple.Block, pos model.BlockPositio
 	if pos == model.Block_Right {
 		columnPos += 1
 	}
-	row.Model().ChildrenIds = slice.Insert(row.Model().ChildrenIds, columnPos, column.Model().Id)
+	s.insertChildrenIds(row.Model(), columnPos, column.Model().Id)
 	s.changesStructureIgnoreIds = append(s.changesStructureIgnoreIds, "cd-"+opId, "ct-"+opId, "r-"+opId, row.Model().Id)
 	return
 }
 
 func (s *State) wrapToRow(opId string, parent, b simple.Block) (row simple.Block, err error) {
+	column := s.addNewBlockAndWrapToColumn(opId, b)
+	row = s.addNewColumnToRow(opId, column)
+	pos := slice.FindPos(parent.Model().ChildrenIds, b.Model().Id)
+	if pos == -1 {
+		return nil, fmt.Errorf("creating row: can't find child[%s] in given parent[%s]", b.Model().Id, parent.Model().Id)
+	}
+	parent.Model().ChildrenIds[pos] = row.Model().Id
+	return
+}
+
+func (s *State) setParent(parent *model.Block, childrenIds []string) {
+	s.cacheParentUntilSame(parent, childrenIds, false)
+}
+
+func (s *State) cacheParentUntilSame(parent *model.Block, childrenIds []string, untilSame bool) {
+	parent.ChildrenIds = childrenIds
+	if s.isIdsCacheInited() {
+		cache := s.getSubIdsCache()
+		for _, childId := range childrenIds {
+			if untilSame {
+				if curParentId, ok := cache[childId]; ok && curParentId == parent.Id {
+					break
+				}
+			}
+			cache[childId] = parent.Id
+		}
+	}
+}
+
+func (s *State) appendChildrenIdsAtStart(block *model.Block, ids ...string) {
+	s.setParent(block, append(block.ChildrenIds, ids...))
```

Comment:
I advise you to use slices.Concat because of https://github.com/go-critic/go-critic/issues/865

Maybe it's too cautious, but why not to go safer here?